app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -0,0 +1,262 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Heatmap from "~/components/visualizations/heatmap/Heatmap";
+import { getAMRCounts } from "~/api/amr";
+
+import cs from "./amr_heatmap_vis.scss";
+
+const METRIC_ALLELES = "alleles";
+const METRIC_GENES = "genes";
+
+export default class AMRHeatmapVis extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      viewLevel: "genes",
+      metric: "coverage",
+    };
+
+    this.heatmap = null;
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  componentDidUpdate() {
+    let rows = this.createHeatmapLabels();
+    let columns = this.state.samples;
+    let values = this.computeHeatmapValues(rows);
+    let options = {};
+    this.renderHeatmap(rows, columns, values, options);
+  }
+
+  async requestAMRCountsData(sampleIds) {
+    let rawResponse = await getAMRCounts(sampleIds);
+    let [geneData, samples] = this.extractData(rawResponse);
+    let alleleToGeneMap = this.mapAllelesToGenes(geneData);
+    this.setState((state, props) => {
+      return {
+        rawData: rawResponse,
+        geneData: geneData,
+        samples: samples,
+        alleleToGeneMap: alleleToGeneMap,
+      };
+    });
+  }
+
+  extractData(rawData) {
```

Comment:
Instead of calling it `rawData`, I would call it something like `rawSamplesWithAMRCounts`. Gives a little more context to what that variable is when reading the code. Similarly, I would rename `extractData`. In general, try to be a bit more specific with naming.